### PR TITLE
fix: Add CODE_SNIPPET_USE_AUTO_CONFIG environment variable

### PR DESCRIPTION
**Pull request by ddl-rliu**
# TL;DR
The inputs/outputs Python code snippet leads to a "Failed to connect" error

Introduce the environment variable `CODE_SNIPPET_USE_AUTO_CONFIG`, to control whether the config object in the inputs/outputs Python code snippet is automatically constructed i.e. `Config.auto()`. This will tend to use the `admin.endpoint` that users can set in their `~/.flyte/config.yaml`

It can be left unset to keep [the current behavior e.g. uses `Config.for_endpoint(window.location.host)`
](https://github.com/flyteorg/flyteconsole/blob/7636e32d2b1dbefb7d08fb79e5ed063c3ad44ca4/packages/oss-console/src/components/Executions/ExecutionDetails/ExecutionNodeURL.tsx#L39)
## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Users already configure flytekit in `~/.flyte/config.yaml` to set `admin.endpoint` to the correct gRPC endpoint. We can leverage this by using `Config.auto()` auto config, and building the code snippet around this Config.

Tested fix correctly shows the new code snippet when setting is enabled:
image

Tested fix uses current behavior when setting is not enabled:
image

## Tracking Issue
https://github.com/flyteorg/flyte/issues/5578

### CODE REVIEW

File: packages/oss-console/src/components/Executions/ExecutionDetails/ExecutionNodeURL.tsx [hunk 1]
```diff
@@ -31,19 +32,18 @@ export const ExecutionNodeURL: React.FC<{
   const isHttps = /^https:/.test(window.location.href);
   const ref = React.useRef<HTMLDivElement>(null);
 
-  const code = isHttps
-    ? // https snippet
-      `from flytekit.remote.remote import FlyteRemote
+  const config =
+    env.CODE_SNIPPET_USE_AUTO_CONFIG === "true"
+      ? 'Config.auto()'
+      : isHttps
```
[the remark this answers is not in the document]
Comment (ddl-rliu) [reply packages/oss-console/src/components/Executions/ExecutionDetails/ExecutionNodeURL.tsx:38]: For context, this follows the nested ternary pattern that I've seen e.g. here https://github.com/flyteorg/flyteconsole/blob/7636e32d2b1dbefb7d08fb79e5ed063c3ad44ca4/packages/oss-console/src/components/flytegraph/ReactFlow/transformDAGToReactFlowV2.tsx#L199-L205

Since python booleans aren't one-to-one with JS booleans, it would have to look like
```js
const insecure = isHttps ? "False" : "True"
 ... `Config.for_endpoint("${window.location.host}",${insecure})`
```
which I wasn't sure would be much clearer.

[the remark this answers is not in the document]
Comment (ddl-rliu) [reply packages/oss-console/src/components/Executions/ExecutionDetails/ExecutionNodeURL.tsx:39]: Gotcha! Okay, updated it here: https://github.com/flyteorg/flyteconsole/pull/883/commits/0dfb7b84fd6f9b6ae0d8ed72c5894e856d2e3d20

**Review by eapolinario — APPROVED** [review]
